src/LinAlg/hiopVectorRajaPar.cpp

```diff
@@ -218,6 +225,29 @@ void hiopVectorRajaPar::setToConstant(double c)
     });
 }
 
+void hiopVectorRajaPar::set_to_random_uniform(double minv, double maxv)
+{
+  double* data = data_dev_;
+#ifdef HIOP_USE_GPU
+  if(mem_space_ == "DEVICE") {
+    hiop::device::array_random_uniform_kernel(n_local_, data, minv, maxv);
+  } else {
+    // TODO: unified memory needs to use host implementation or cuda api. Why? 
+    hiop::device::array_random_uniform_kernel(n_local_, data, minv, maxv);  // this fails
+
+//    hiop::host::array_random_uniform_kernel(n_local_, data, minv, maxv);  // this works
+
+    // this 3-line method from cuda api also works
+//  hiop::device::array_random_uniform_kernel(n_local_, data);
+//  scale(maxv-minv);
+//  addConstant(minv);
+  }
+#else
+  // TODO: add function for openmp polocy
+  hiop::host::array_random_uniform_kernel(n_local_, data, minv, maxv);
+#endif
```

Comment:
Unit test fails when we use unified memory. Not sure why. move to offline discussion.

Reply:
for the record, lack of device synchronization was causing this.